src/dstack/_internal/server/services/backends/configurators/cudo.py

```diff
@@ -0,0 +1,81 @@
+import json
+from typing import List
+
+from dstack._internal.core.backends.base import Backend
+from dstack._internal.core.backends.cudo import CudoBackend, CudoConfig
+from dstack._internal.core.models.backends.base import (
+    BackendType,
+    ConfigElementValue,
+    ConfigMultiElement,
+)
+from dstack._internal.core.models.backends.cudo import (
+    CudoConfigInfo,
+    CudoConfigInfoWithCreds,
+    CudoConfigInfoWithCredsPartial,
+    CudoConfigValues,
+    CudoCreds,
+    CudoStoredConfig,
+)
+from dstack._internal.server.models import BackendModel, ProjectModel
+from dstack._internal.server.services.backends import Configurator
+
+REGIONS = [
+    "no-luster-1",
+    "se-smedjebacken-1",
+    "gb-london-1",
+    "se-stockholm-1",
+    "us-newyork-1",
+    "us-santaclara-1",
+]
+
+DEFAULT_REGION = "no-luster-1"
+
+
+class CudoConfigurator(Configurator):
+    TYPE: BackendType = BackendType.CUDO
+
+    def get_default_configs(self) -> List[CudoConfigInfoWithCreds]:
+        return []
+
+    def get_config_values(self, config: CudoConfigInfoWithCredsPartial) -> CudoConfigValues:
+        config_values = CudoConfigValues()
+        if config.creds is None:
+            return config_values
+        config_values.regions = self._get_regions_element(
+            selected=config.regions or [DEFAULT_REGION]
+        )
+        return config_values
```

Comment:
There should be a basic validation of cloud credentials. You can do it by calling Cudo API in get_config_values() and raising BackendInvalidCredentialsError() if the API call fails. This way, uses will get errors/warnings upon backend creation.

Reply:
Thank you. I will modify accordingly

Reply:
Completed and tested
<img width="1347" alt="Screenshot 2024-02-26 at 15 15 37" src="https://github.com/dstackai/dstack/assets/3397983/bca29e43-7e1c-4f81-a458-0f28ee691cac">